src/context/AuthContext.jsx

```diff
@@ -0,0 +1,65 @@
+import axios from 'axios';
+import { createContext, useContext, useState, useEffect } from 'react';
+import { useNavigate } from 'react-router';
+
+import { API_ENDPOINT } from '../config/config.js';
+
+const AuthContext = createContext({
+  currentUser: { identifier: '', password: '' },
+  isUserLoggedIn: false,
+  handleLogin: () => {},
+  handleLogout: () => {},
+});
+
+export const useAuthContext = () => useContext(AuthContext);
+
+export const AuthContextProvider = ({ children }) => {
+  const [currentUser, setCurrentUser] = useState(null);
+  const [isUserLoggedIn, setUserLoggedIn] = useState(false);
+  const navigate = useNavigate();
+
+  const handleLogin = async (identifier, password) => {
+    const user = { identifier, password };
+
+    try {
+      const response = await axios.post(API_ENDPOINT, user);
```

Comment:
This should be moved to `auth` service, similar to what was discussed above for the `register` functionality